### PR TITLE
Add SearchResponse. Refactor doc and example JSON-LD.

### PR DESCRIPTION
**Pull request by arwhyte**
This PR refactors the Search Profile.  It adds a generated `SearchResponse` (subtype of  `Entity`) which extends `Entity` with the following properties:

searchProvider \<SoftwareApplication\>
resourceSearched \<DigitalResource | SoftwareApplication\>
query \<Query\>
searchResultsItemCount \<integer\>
searchResults \<DigitalResourceCollection\>

The `Query` (subtype of `Entity`) extends `Entity` with the following properties:

creator \<Person\>
searchTerms \<string\>

The refactoring is based in part on Caliper Working Group discussions plus my own reconsidering of earlier drafts of the profile.  It makes an attempt to retain the option of including search results which is desired by some though not all working group members. It also draws a certain inspiration from the [OpenSearch 1.1 draft](https://github.com/dewitt/opensearch/blob/master/opensearch-1-1-draft-6.md) which Google draws upon for its [custom search API](https://developers.google.com/custom-search/json-api/v1/overview).

Note that the images will need updating.  I will tweek those in a followup commit to this PR.

### CODE REVIEW

File: profiles/search/caliper-profile-search-v1p1.html [hunk 1]
```diff
@@ -331,34 +332,42 @@ <h3>Query</h3>
...
-            >Person</a> | <a href=
-                "https://www.imsglobal.org/sites/default/files/caliper/v1p1/caliper-spec-v1p1/caliper-spec-v1p1.html#iriDef"
+               "https://www.imsglobal.org/sites/default/files/caliper/v1p1/caliper-spec-v1p1/caliper-spec-v1p1.html#entity"
+            >SoftwareApplication</a> | <a href=
+               "https://www.imsglobal.org/sites/default/files/caliper/v1p1/caliper-spec-v1p1/caliper-spec-v1p1.html#iriDef"
             >IRI</a></td>
-            <td>The <code>Person</code> who authored or submitted the <code>Query</code>. The
-              <code>creator</code> value MUST be expressed either as an object or as a string
-              corresponding to the creator's IRI.</td>
+            <td>The <code>SoftwareApplication</code> that is the provider of this
```
Comment (ViktorHaag) [inline profiles/search/caliper-profile-search-v1p1.html]: So, if I understand this model, given (say) Google as the UX, then the SearchProvider is Google, and the SearchSTring is the thing you type into the search box, and the SearchResponse is the page that results when you hit Go after typing in your search string?

File: profiles/search/caliper-profile-search-v1p1.html [hunk 2]
```diff
@@ -331,34 +332,42 @@ <h3>Query</h3>
...
+              <code>SearchResponse</code>. The <code>searchProvider</code> value MUST be expressed
```
Comment (ViktorHaag) [inline profiles/search/caliper-profile-search-v1p1.html]: mismatch between search "response" and search "provider"?

File: profiles/search/caliper-profile-search-v1p1.html [hunk 3]
```diff
@@ -331,34 +332,42 @@ <h3>Query</h3>
...
+            <td>resourceSearched</td>
             <td><a href=
               "https://www.imsglobal.org/sites/default/files/caliper/v1p1/caliper-spec-v1p1/caliper-spec-v1p1.html#entity"
             >Entity</a> | <a href=
                "https://www.imsglobal.org/sites/default/files/caliper/v1p1/caliper-spec-v1p1/caliper-spec-v1p1.html#iriDef"
             >IRI</a></td>
             <td>The <code>Entity</code>, typically a <code>DigitalResource</code> or
               <code>SoftwareApplication</code>, that is the target of the search. The
-              <code>searched</code> value MUST be expressed either as an object or as a string
+              <code>resourceSearched</code> value MUST be expressed either as an object or as a
```
Comment (ViktorHaag) [inline profiles/search/caliper-profile-search-v1p1.html]: How do we see a difference between the search provider and the resource searched? Is like a library search where you have a UX that acts as your "search manager" application and it lets you check boxes for which pools of data you want to search within (the pools of data being the "resource searched")? If that's the use case, should this be a list of resources searched (i.e. I type in a single string, choose some resources to search, and then hit Go, and the provider waits for results and then generates a single search result event that describes this activity).

File: profiles/search/caliper-profile-search-v1p1.html [hunk 4]
```diff
@@ -231,11 +231,11 @@ <h3>SearchEvent</h3>
           </tr>
           <tr>
             <td>generated</td>
-            <td><a href="#query">Query</a> | <a href=
+            <td><a href="#query">SearchResponse</a> | <a href=
```
Comment (ViktorHaag) [inline profiles/search/caliper-profile-search-v1p1.html]: We may have confusion between _response_ for searches and _results_ for searches? We just need to clearly describe these, I guess?

_Response_ is the search provider's presentation of a list of results after it searches a bunch of resources with your provided search string?

_Result_ is the data returned by a resource-searched with your search string that gets formed into the provider's "response"?

File: profiles/search/caliper-profile-search-v1p1.html (hunk 2 above)
Comment (arwhyte) [reply profiles/search/caliper-profile-search-v1p1.html]: searchProvider description: 
> The `SoftwareApplication` that is the provider of this `SearchResponse`. The `searchProvider` value MUST be expressed either as an object or as a string corresponding to the resources's IRI.

That reads correctly to me.

Comment (ViktorHaag) [reply profiles/search/caliper-profile-search-v1p1.html]: Yeah, just hard to parse all this HTML. 8(

**Comment by arwhyte** [comment]
PR updated.
1.  searchResults type changed from object (`DigitalResourceCollection`) to array<`DigitalResource`>.
2.  images updated.
3. typos and misc. errors fixed.

Raw git view of updated profile:

https://rawgit.com/IMSGlobal/caliper-spec/4fdd3bc42bb41f4499a4b7fe9ae46103a3049867/profiles/search/caliper-profile-search-v1p1.html

**Comment by arwhyte** [comment]
If a learner searches a `DigitalResource` for a particular term, the `SearchResponse.searchResults [\<`DigitalResource`\>] could utilize the `Frame` Entity (sub type `DigitalResource`) to provide an indexed location for matched terms in the text.

File: profiles/search/caliper-profile-search-v1p1.html (hunk 1 above)
Comment (arwhyte) [reply profiles/search/caliper-profile-search-v1p1.html]: I've changed `SearchResponse.searchResults` value from a map to an array [\<`DigitalResource`\>] which should address your first concern.  As for the second, what is the value from a learning analytics perspective pegging the precise instant that a learner submitted their query vs describing the results, if any, of the query they submitted?

File: profiles/search/caliper-profile-search-v1p1.html (hunk 3 above)
Comment (arwhyte) [reply profiles/search/caliper-profile-search-v1p1.html]: The current model attempts to distinguish between the "engine/provider" and the target of the search (e.g., a catalog, a single `DigitalResource`, etc.) for entity describe purposes.  I would think `Event.edapp` = `SearchResponse.searchProvider` or SearchResponse.provider (slim down the name).

File: profiles/search/caliper-profile-search-v1p1.html (hunk 4 above)
Comment (arwhyte) [reply profiles/search/caliper-profile-search-v1p1.html]: Yep.  Distinguishing between the two allows one to describe a response that returns zero results, which may well have significance from a learning standpoint when evaluating the filtering criteria.
